apps/files/js/newfilemenu.js

```diff
@@ -203,7 +203,8 @@
 				iconClass: actionSpec.iconClass,
 				fileType: actionSpec.fileType,
 				actionHandler: actionSpec.actionHandler,
-				checkFilename: actionSpec.checkFilename
+				checkFilename: actionSpec.checkFilename,
+				shouldShow: actionSpec.shouldShow || (() => true)
```

Comment:
```suggestion
				shouldShow: actionSpec.shouldShow,
```

Should be enough as the filter would still include it if it is undefined

Reply:
The render function looks for `showMenu: true` in order to consider adding the menu item so I think we need this.

Reply:
Just tested and the suggestion works as the undefined value will evaluate to true with `!item.showMenu` which means that the item is always shown if no showMenu is passed.